src/components/VideoPlayer/VideoPlayer.tsx

```diff
@@ -37,13 +47,32 @@ type progressState = {
 class VideoPlayer extends React.Component<VideoPlayerProps, VideoPlayerState> {
   player: ReactPlayer | null = null;
   state: VideoPlayerState = {
+    ignoreUpdates: true,
     playing: false,
     seeking: false,
     played: 0,
     url: "",
     videoUrlQueue: [],
   };
 
+  constructor(props: VideoPlayerProps) {
+    super(props);
+    const queryString = '?' + props.router.asPath.split('?')[1];
+    const code = new URLSearchParams(queryString).get('code');
```

Comment:
```suggestion
    const { code } = props.router.query;
```
Not sure why, but the above snippet doesn't work. Maybe because I'm refreshing the page?

Reply:
~~Use the `useRouter()` hook:~~

```typescript
  const {
    query: { code },
  } = useRouter();
  const {
  const codestr = code?.toString();
```

Just realized that this is class component so probably won't work. You could just get the code from [pages/room.tsx: line 14](https://github.com/Galactus-Player/web/blob/main/src/pages/room.tsx#L14) and pass the code down as prop to `VideoPlayer`. Just add something like `roomCode` in `VideoPlayerProps`, and do `<VideoPlayer roomCode={roomCode}/>` in [pages/room.tsx](https://github.com/Galactus-Player/web/blob/main/src/pages/room.tsx#L29).